src/test/scala/PerftTest.scala

```diff
@@ -2,128 +2,23 @@ package chess
-    }
-    "gentest-3283" in {
-      val game =
-        Game(
-          Option(Chess960),
-          Option(EpdFen("r1bqkb1r/p1pp1ppp/1p3n2/6B1/Pn1p2P1/7B/RPP1PP1P/1N1QK1NR w Kkq -"))
-        )
-      perft(game, 3) must be equalTo 26302
-    }
-    "gentest-3523" in {
-      val game = Game(Option(Chess960), Option(EpdFen("1n1K4/3bB3/2k5/2p5/8/8/3p4/8 w - -")))
-      perft(game, 3) must be equalTo 933
-    }
-    "gentest-4253" in {
-      val game = Game(Option(Chess960), Option(EpdFen("2Bk4/2N4N/1Q6/P2pb1P1/1R5p/2r1p2P/4K1R1/8 b - -")))
-      perft(game, 3) must be equalTo 17116
-    }
-    "gentest-4309" in {
-      val game =
-        Game(Option(Chess960), Option(EpdFen("1nb1qbn1/4rk1r/ppNp3p/5ppP/P5P1/1P3P2/2PPP1RN/1R1QKB2 b - -")))
-      perft(game, 3) must be equalTo 20587
-    }
-    "gentest-5008" in {
-      val game = Game(Option(Chess960), Option(EpdFen("5nk1/7p/r1p4b/p5PP/K4p2/3R4/5B2/8 w - -")))
-      perft(game, 3) must be equalTo 8214
-    }
-    "gentest-5569" in {
-      val game =
-        Game(Option(Chess960), Option(EpdFen("r3k1n1/p2p2p1/1p4N1/2r2pq1/2b2PPp/b1p5/P3PB1P/RQ1K1B1R b q -")))
-      perft(game, 3) must be equalTo 54944
-    }
-    "gentest-6195" in {
-      val game =
-        Game(Option(Chess960), Option(EpdFen("Bnbqk2r/2pppp2/pp4p1/7p/P2b2Q1/2P1P3/1P1P1PPP/RNB2KNR w k -")))
-      perft(game, 3) must be equalTo 41002
+  private def genTests(name: String, tests: List[PerftTestCase]): Fragments =
+    name should {
+      Fragments.foreach(tests) { pts =>
+        pts.id should {
+          val result = pts.calculate()
+          Fragments.foreach(result) { r =>
+            s"${r.depth}" in {
+              r.result must be equalTo r.expected
+            }
+          }
+        }
+      }
     }
-    "gentest-6689" in {
-      val game = Game(Option(Chess960), Option(EpdFen("8/k5b1/7N/2b5/8/1K3N2/8/4R3 w - -")))
-      perft(game, 3) must be equalTo 17996
-    }
-  }
-
-  "calculate chess960 perfts" should {
-    // source: http://www.talkchess.com/forum/viewtopic.php?t=55274
-    "x-fen 00" in {
-      val game =
-        Game(Option(Chess960), Option(EpdFen("r1k1r2q/p1ppp1pp/8/8/8/8/P1PPP1PP/R1K1R2Q w KQkq - 0 1")))
-      perft(game, 3) must be equalTo 12333
-    }
-    "x-fen 01" in {
-      val game =
-        Game(Option(Chess960), Option(EpdFen("r1k2r1q/p1ppp1pp/8/8/8/8/P1PPP1PP/R1K2R1Q w KQkq - 0 1")))
-      perft(game, 3) must be equalTo 20218
-    }
-    "x-fen 02" in {
-      val game = Game(Option(Chess960), Option(EpdFen("8/8/8/4B2b/6nN/8/5P2/2R1K2k w Q - 0 1")))
-      perft(game, 4) must be equalTo 118388
-    }
-    "x-fen 03" in {
-      val game = Game(Option(Chess960), Option(EpdFen("2r5/8/8/8/8/8/6PP/k2KR3 w K - 0 1")))
-      perft(game, 4) must be equalTo 57700
-    }
-    "x-fen 04" in {
-      val game = Game(Option(Chess960), Option(EpdFen("4r3/3k4/8/8/8/8/6PP/qR1K1R2 w KQ - 0 1")))
-      perft(game, 3) must be equalTo 12858
-    }
-  }
 
```

Comment:
Now the tests are a bit weird. It'll calculate all the results before assert results. But I think it's okay.